package/MDAnalysis/coordinates/IMD.py

```diff
@@ -0,0 +1,256 @@
+    in memory as data is received from the socket and awaits reading by the client. For analyses
+    that periodically perform heavier computation at fixed intervals, say for example once every
+    200 received frames, increasing this value will decrease the amount of time the simulation
+    engine spends in a paused state and potentially decrease total analysis time, but will require
+    more RAM.
+
+    Parameters
+    ----------
+    filename : a string of the form "imd://host:port" where host is the hostname
+        or IP address of the listening simulation engine's IMD server and port
+        is the port number.
+    n_atoms : int (optional)
+        number of atoms in the system. defaults to number of atoms
+        in the topology. Don't set this unless you know what you're doing.
+    buffer_size: int (optional) default=10*(1024**2)
+        number of bytes of memory to allocate to the :class:`IMDClient`'s
+        internal buffer. Defaults to 10 megabytes.
+    kwargs : dict (optional)
+        keyword arguments passed to the constructed :class:`IMDClient`
+    """
+
+    format = "IMD"
+
+    @store_init_arguments
+    def __init__(
+        self,
+        filename,
+        n_atoms=None,
+        buffer_size=10 * (1024**2),
+        **kwargs,
+    ):
+        if not HAS_IMDCLIENT:
+            raise ImportError(
+                "IMDReader requires the imdclient package. "
+                "Please install it with 'pip install imdclient'."
+            )
+
+        super(IMDReader, self).__init__(filename, **kwargs)
+
+        self._imdclient = None
+        logger.debug("IMDReader initializing")
+
+        if n_atoms is None:
+            raise ValueError("IMDReader: n_atoms must be specified")
+        self.n_atoms = n_atoms
+
+        try:
+            host, port = parse_host_port(filename)
+        except ValueError as e:
+            raise ValueError(f"IMDReader: Invalid IMD URL '{filename}': {e}")
+
+        # This starts the simulation
+        self._imdclient = IMDClient(
+            host, port, n_atoms, buffer_size=buffer_size, **kwargs
+        )
+
+        imdsinfo = self._imdclient.get_imdsessioninfo()
+        if imdsinfo.version != 3:
+            raise NotImplementedError(
+                f"IMDReader: Detected IMD version v{imdsinfo.version}, "
+                + "but IMDReader is only compatible with v3"
+            )
+
+        self.ts = self._Timestep(
+            self.n_atoms,
+            positions=imdsinfo.positions,
+            velocities=imdsinfo.velocities,
+            forces=imdsinfo.forces,
+            **self._ts_kwargs,
+        )
+
+        try:
+            self._read_next_timestep()
+        except EOFError as e:
+            raise RuntimeError(f"IMDReader: Read error: {e}") from e
+
+    def _read_frame(self, frame):
+
+        try:
+            imdf = self._imdclient.get_imdframe()
+        except EOFError as e:
+            raise e
```

Comment:
What's the point of running this in try/except, given that it just raises the exception that the code already raises? Can this just be
```suggestion
        imdf = self._imdclient.get_imdframe()
```
What am I missing here?

Reply:
You're right, cleaned up `try/except`